### PR TITLE
scheduler: Implement host sync and checking

### LINKED ISSUE #1921: scheduler: Reconnect host event streams intelligently
Currently when a formation, host, and leader event stream gets disconnected, the scheduler attempts to reconnect. We need to do the same with job streams from a host instead of immediately giving up on the host.

As part of this, also remove the automatic host unfollow that happens if we can't connect to a host client when scheduling a job, which will involve needing to pick a different host for the job.

Also figure out how to correctly handle stopping a job when we can't connect to the host running the job to be stopped.

### PR DESCRIPTION
**Pull request by lmars**
Closes #1921.

Fixes part of #2004.

### CODE REVIEW

File: controller/scheduler/scheduler.go [hunk 1]
```diff
@@ -1075,6 +1120,16 @@ func (s *Scheduler) tickSyncFormations(d time.Duration) {
 	}()
 }
 
+func (s *Scheduler) tickSyncHosts(d time.Duration) {
+	logger.Info("starting sync hosts ticker", "duration", d)
+	go func() {
+		ch := time.Tick(d)
+		for range ch {
```
Comment (titanous) [inline controller/scheduler/scheduler.go]: No need to assign to `ch`, could just put it inline.

**Comment by titanous** [comment]
LGTM. What did you do to test this?

**Comment by lmars** [comment]
@titanous a very crude test, but I added the following code to `flynn-host` to temporarily mark it as down:

```go
go func() {
	for range time.Tick(30 * time.Second) {
		d.mtx.Lock()
		d.hb.Close()
		time.Sleep(7 * time.Second)
		d.hb, _ = disc.AddServiceAndRegisterInstance("flynn-host", d.inst)
		d.mtx.Unlock()
	}
}()
```

I then observed the following in the scheduler log:

```
t=2015-10-29T21:07:24+0000 lvl=info msg="handling host event" component=scheduler fn=HandleHostEvent event.type=down
t=2015-10-29T21:07:24+0000 lvl=warn msg="host service is down, marking as unhealthy and triggering host checks" component=scheduler host.id=host
t=2015-10-29T21:07:24+0000 lvl=info msg="triggering host checks" component=scheduler
t=2015-10-29T21:07:24+0000 lvl=info msg="performing host checks" component=scheduler fn=PerformHostChecks
t=2015-10-29T21:07:24+0000 lvl=info msg="getting status of unhealthy host" component=scheduler fn=PerformHostChecks host.id=host
t=2015-10-29T21:07:24+0000 lvl=info msg="host is now healthy" component=scheduler fn=PerformHostChecks host.id=host
t=2015-10-29T21:07:31+0000 lvl=info msg="handling host event" component=scheduler fn=HandleHostEvent event.type=up
t=2015-10-29T21:07:31+0000 lvl=info msg="host is up, starting job event stream" component=scheduler fn=HandleHostEvent event.type=up host.id=host
t=2015-10-29T21:07:32+0000 lvl=info msg="triggering host sync" component=scheduler
t=2015-10-29T21:07:32+0000 lvl=info msg="syncing hosts" component=scheduler fn=SyncHosts
```

So the host check actually passed (as the host was still responsive) and nothing untoward happened.

I am investigating the CI failures, but I think they may be unrelated.

**Comment by lmars** [comment]
The failures were unrelated, merging.

**Pull request merged**